server/src/main/java/org/apache/uniffle/server/buffer/ShuffleBufferManager.java

```diff
@@ -362,10 +363,12 @@ public boolean requireReadMemoryWithRetry(long size) {
   public void releaseReadMemory(long size) {
     if (readDataMemory.get() >= size) {
       readDataMemory.addAndGet(-size);
+      ShuffleServerMetrics.gaugeReadBufferUsedSize.dec(size);
```

Comment:
```suggestion
      ShuffleServerMetrics.gaugeReadBufferUsedSize.set(readDataMemory.get());
```
How about this change?

Reply:
Yes, it's Ok. The metrics should be always consistent with the value of `readDataMemory.`

Reply:
> In my opinion, the suggest might not be better than the previous one.
> 
> Consider this case:
> 
> ```
> T1, T2 for two threads.
> t0, t1, t2 for times
> N: current read buffer
> t0: T1: readDataMemory.addAndGet(-size)
> t1: T1: readDataMemory.get() --> N - size
> t2: T2: readDataMemory.addAndGet(-size)
> t3: T2: readDataMemory.get() --> N - size - size
> t4: T2 : gaugeReadBufferUsedSize <- N -size - size
> t5: T1:  gaugeReadBufferUsedSize  <- N - size
> ```
> 
> Unless there's safe bet to actually reset the gauge to the readDataMemory.get().

You are right, My mistake. Forgot that the same problem occurred before. :joy:  #404